### PR TITLE
make infohash and peerID byte arrays

### PR DESCRIPTION
**Pull request by mrd0ll4r**

### CODE REVIEW

**Comment by pushrax** [comment]
It's guaranteed to be a 20-byte string right?

**Comment by joshdekock** [comment]
@pushrax it says 20-byte in the spec, so I assume that all complying clients would use 20-bytes.

**Comment by mrd0ll4r** [comment]
fixed, now says 20-byte string. (We don't actually check/enforce that yet, but we will soon)

**Comment by jzelinskie** [comment]
If it is actually 20 bytes, why can't this type `[20]byte`?

**Comment by mrd0ll4r** [comment]
I kinda want to make it a `[20]byte`, dunno how often we need to convert it to a string - I'll have a look.

File: server/store/middleware/infohash/blacklist_test.go [hunk 1]
```diff
@@ -40,12 +40,26 @@ var mock store.StringStore = &storeMock{
 	strings: make(map[string]struct{}),
 }
 
+var (
+	ih1, ih2 chihaya.InfoHash
+)
+
+func init() {
```
Comment (jzelinskie) [inline server/store/middleware/infohash/blacklist_test.go]: This doesn't need to be in an `init` function. You can use literals on new types derived from arrays: https://play.golang.org/p/s1ISSQ_XMq

**Comment by jzelinskie** [comment]
I really dislike having this all over the place:

```go
id := [20]byte{}
copy(id[:], p.peerID)
```
Can we have a helper function that handles this?

File: chihaya.go [hunk 2]
```diff
@@ -12,10 +12,50 @@ import (
...
+	copy(a[:], b)
+	return PeerID(a)
+}
+
+func PeerIDFromString(s string) PeerID {
+	if len(s) != 20{
+		panic("peer ID must be 20 bytes")
+	}
+
+	a := [20]byte{}
```
Comment (jzelinskie) [inline chihaya.go]: Let's replace this line with `var buf [20]byte`.
For initializing a zero value, var is usually preferred.

File: chihaya.go [hunk 3]
```diff
@@ -12,10 +12,50 @@ import (
...
+	if len(b) != 20 {
+		panic("peer ID must be 20 bytes")
+	}
+
+	a := [20]byte{}
+	copy(a[:], b)
+	return PeerID(a)
+}
+
+func PeerIDFromString(s string) PeerID {
```
Comment (jzelinskie) [inline chihaya.go]: All these new functions should have doc strings.

**Comment by jzelinskie** [comment]
LGTM

**Pull request merged**